sound/soc/sof/intel/hda-dai.c

```diff
@@ -268,7 +266,18 @@ static int hda_dai_prepare(struct snd_pcm_substream *substream, struct snd_soc_d
 	if (ret < 0)
 		return ret;
 
-	return hda_dai_hw_params_update(substream, &rtd->dpcm[stream].hw_params, dai);
+	if (ops && ops->dma_ops && ops->dma_ops->stream_get_dma)
+		hext_stream = ops->dma_ops->stream_get_dma(sdev, dai, substream);
+
+	if (tplg_ops && tplg_ops->dai_config) {
+		unsigned int flags = SOF_DAI_CONFIG_FLAGS_HW_PARAMS;
+
+		flags |= SOF_DAI_CONFIG_FLAGS_2_STEP_STOP << SOF_DAI_CONFIG_FLAGS_QUIRK_SHIFT;
+		data.dai_data = hdac_stream(hext_stream)->stream_tag - 1;
```

Comment:
@jsarha dai_index is used only for SDW and dai_data is used for HDA. I cant initialize it just for the sake of it. Please check the code in ipc4-topology.c to see if it is used at all or not

Reply:
I think it would still make sense to initialize dai_index to some defined invalid value, to make it clear its not to be used. Also, if both values are never used at the same time, then shouldn't this be a union? But for my part, feel free to resolve the conversation (and the one bellow).